support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoMfaProviderFactoryBean.java

```diff
@@ -0,0 +1,57 @@
+package org.apereo.cas.adaptors.duo.authn;
+
+import org.apereo.cas.authentication.MultifactorAuthenticationUtils;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.util.http.HttpClient;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import org.springframework.beans.factory.config.AbstractFactoryBean;
+
+
+/**
+ * Implementation of AbstractFactoryBean used to create Duo provider instances.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+@Slf4j
+@Setter
+public class DuoMfaProviderFactoryBean extends AbstractFactoryBean<DefaultDuoMultifactorAuthenticationProvider> {
+
+    private final HttpClient httpClient;
+
+    private final CasConfigurationProperties casProperties;
+
+    private String duoId;
+
+    public DuoMfaProviderFactoryBean(final HttpClient httpClient,
+                                     final CasConfigurationProperties casProperties) {
+        this.httpClient = httpClient;
+        this.casProperties = casProperties;
+    }
+
+    @Override
+    public DefaultDuoMultifactorAuthenticationProvider createInstance() throws Exception {
+        if (duoId != null) {
```

Comment:
This should be an error; cant create a provider without an id?

Reply:
This relates to the custom bean definition registries in DuoProviderRegistrationConfiguration.  Autoconfigured bean with @RefreshScope is created without a specific Duo instance.  It's BeanDefiintion is then cloned and the a duo instance is passed to it.  

Hopefully some collective brain power came up with a better solution.  I fell in some deep dark holes on this one.